src/main/java/com/zufar/onlinestore/payment/service/PaymentEventHandler.java

```diff
@@ -0,0 +1,44 @@
+package com.zufar.onlinestore.payment.service;
+
+import com.stripe.model.Event;
+import com.stripe.model.PaymentIntent;
+import com.zufar.onlinestore.payment.entity.Payment;
+import com.zufar.onlinestore.payment.enums.PaymentStatus;
+import com.zufar.onlinestore.payment.repository.PaymentRepository;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import java.util.Objects;
+
+@Slf4j
+@RequiredArgsConstructor
+@Service
+public class PaymentEventHandler {
+
+    private final PaymentRepository paymentRepository;
+
+    public void handleEvent(Event event, PaymentIntent paymentIntent) {
+        if (!Objects.isNull(paymentIntent) && !Objects.isNull(event)) {
```

Comment:
May be it would be easier to read this way?
```
if (Objects.isNull(paymentIntent) || Objects.isNull(event)) {
  // add some log.info() here, please
  return;
}
```

Reply:
done